relay/writer/file_util.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	uuid "github.com/satori/go.uuid"
+	gmysql "github.com/siddontang/go-mysql/mysql"
+	"github.com/siddontang/go-mysql/replication"
+
+	"github.com/pingcap/dm/pkg/binlog/event"
+	"github.com/pingcap/dm/pkg/binlog/reader"
+	"github.com/pingcap/dm/pkg/gtid"
+	"github.com/pingcap/dm/relay/common"
+)
+
+// checkBinlogHeaderExist checks if the file has a binlog file header.
+func checkBinlogHeaderExist(filename string) (bool, error) {
```

Comment:
can we split it into two functions to save open file operation, like in `checkFormatDescriptionEventExist`

1. checkBinlogHeaderExist(filename string)
2. checkBinlogHeaderExist(fd *File)

Reply:
👍 splited